data/sea/52-psv.h

```diff
@@ -107,7 +122,75 @@ static int INLINE psv_compare (const char *xs, size_t xs_size, const char *ys, s
     return xs_size - ys_size;
 }
 
-static ierror_loc_t psv_read_buffer (psv_state_t *s)
+/* Seek to a line where the entity changes, or EOF */
+static int INLINE psv_skip_to_next_entity (psv_state_t *s)
+{
+    const char  *end_ptr    = s->input_ptr + s->input_size;
+    const char  *start_ptr  = end_ptr - s->input_remaining;
+    const char  *line_ptr   = start_ptr;
+    const char  *n_ptr      = 0;
+    const char  *drop_me    = s->entity_dropped;
+    const size_t drop_size  = s->entity_dropped_size;
+
+    for (;;) {
+        const size_t bytes_remaining = end_ptr - line_ptr;
+
+        s->input_ptr = line_ptr;
+        s->input_remaining = bytes_remaining;
+
+        n_ptr = memchr (line_ptr, '\n', bytes_remaining);
+
+        if (n_ptr == 0) {
+            return 1;
+        }
+
+        const char  *entity_ptr  = line_ptr;
+        const char  *entity_end  = memchr (entity_ptr, '|', n_ptr - entity_ptr);
+        const size_t entity_size = entity_end - entity_ptr;
+        const int    new_entity  =  (drop_size != entity_size)
+                                 || (psv_compare ( drop_me, drop_size
+                                                 , entity_ptr, entity_size ) != 0);
+
+        if (new_entity) {
+            /* start the input buffer here */
+            return 0;
+        } else {
+            /* just skip this fact */
+            line_ptr = n_ptr + 1;
```

Comment:
would it be a big change to count the number of dropped facts, and output that to the drop file at the end? just an idea